src/Tokenizer/Token.php

```diff
@@ -166,11 +166,11 @@ public function equals($other, $caseSensitive = true)
             return $this->content === $otherPrototype;
         }
 
-        if (array_key_exists(0, $otherPrototype) && $this->id !== $otherPrototype[0]) {
+        if ((isset($otherPrototype[0]) || array_key_exists(0, $otherPrototype)) && $this->id !== $otherPrototype[0]) {
```

Comment:
I'm wondering now... is there a case where index `0` does not exist ?

Reply:
I dont know too, therefore I played it safe for BC reasons

Reply:
If I would be sure then I would not say that "I'm wondering" :)

we do need check for index 1, as it's optional, one can call this method with `[type, value]` or `[type]`.

try to drop condition for index 0 (always compare the values under 0-index) and see if any test is failing ;)

Reply:
Looks pretty good so far :-)